storage_service/locations/models/pipeline.py

```
@@ -126,3 +127,53 @@ def create_default_locations(self, shared_path=None):
                     p['purpose'], location, self)
                 LocationPipeline.objects.get_or_create(
                     pipeline=self, location=location)
+
+    # HTTP API CALLS
+
+    def _request_api(self, method, path, **kwargs):
+        url = 'http://{}/api/{}'.format(
+            self.remote_name,
+            path.lstrip('/'),
+        )
+        fields = {
+            'username': self.api_username,
+            'api_key': self.api_key
+        }
+        if 'fields' in kwargs:
+            fields.update(kwargs['fields'])
+
+        LOGGER.debug('URL: %s; fields: %s', url, fields)
+        try:
+            response = requests.request(method, url, data=fields, allow_redirects=True)
+        except requests.exceptions.RequestException as e:
+            LOGGER.exception('Unable to connect to pipeline %s', self)
+            raise e
+        LOGGER.debug('Response: %s %s', response.status_code, response.text)
+        return response
+
+    def get_processing_config(self, name):
+        """Obtain a processing configuration XML document given its name. The
+        content is returned as a string, which may be empty in case of errors.
+        """
+        if not name:
+            raise ValueError()
+        url = 'processing-configuration/' + name
+        response = None
+        try:
+            response = self._request_api('GET', url)
+        except requests.exceptions.RequestException:
+            return ''
```

Raise an exception instead of returning an empty string, maybe?